types/Batch/Batch.d.ts

```diff
@@ -0,0 +1,118 @@
+import { IDatedObject, IObjectWithId } from '../base';
+import { IPickup } from '../Pickup';
+import { IScanForm } from '../ScanForm';
+import { LabelFormat } from '../Shipment';
+import { IBatchCreateParameters } from './BatchCreateParameters';
+import { IBatchShipment } from './BatchShipment';
+import { TBatchState } from './BatchState';
+import { TBatchStatuses } from './BatchStatuses';
+
+/**
+ * The Batch object allows you to perform operations on multiple Shipments at once.
+ * This includes scheduling a Pickup, creating a ScanForm and consolidating labels.
+ * Operations performed on Batches are asynchronous and take advantage of our webhook infrastructure.
+ *
+ * @see https://www.easypost.com/docs/api/node#batch-object
+ */
+export declare interface IBatch extends IObjectWithId<'Batch'>, IDatedObject {
+  /**
+   * An optional field that may be used in place of ID in some API endpoints
+   */
+  reference?: string | null;
+
+  /**
+   * The overall state. Possible values are "creating", "creation_failed", "created", "purchasing", "purchase_failed", "purchased", "label_generating", and "label_generated"
+   */
+  state: TBatchState;
+
+  /**
+   * The number of shipments added
+   */
+  num_shipments: number;
+
+  shipments: IBatchShipment[];
+
+  /**
+   * A map of BatchShipment statuses to the count of BatchShipments with that status. Valid statuses are "postage_purchased", "postage_purchase_failed", "queued_for_purchase", and "creation_failed"
+   */
+  status: TBatchStatuses;
+
+  /**
+   * The label image url
+   */
+  label_url?: string | null;
+
+  /**
+   * The created ScanForm
+   */
+  scan_form: IScanForm;
+
+  /**
+   * The created Pickup
+   */
+  pickup: IPickup;
+}
+
+export declare class Batch implements IBatch {
+  public constructor(input: IBatchCreateParameters);
+
+  id: string;
+  mode: 'test' | 'production';
+  object: 'Batch';
+  reference?: string | null;
+  state: TBatchState;
+  num_shipments: number;
+  shipments: IBatchShipment[];
+  status: TBatchStatuses;
+  label_url?: string | null;
+  scan_form: IScanForm;
+  pickup: IPickup;
+  created_at: string;
+  updated_at: string;
+
+  /**
+   * A Batch can be created with or without Shipments.
+   * When created with Shipments the initial state will be creating.
+   * Once the state changes to created a webhook Event will be sent.
+   * When created with no Shipments the initial state will be created and webhook will be sent.
+   *
+   * @see https://www.easypost.com/docs/api/node#create-a-batch
+   */
+  public save(): Promise<Batch>;
+
+  static retrieve(batchId: string): Promise<Batch>;
```

Comment:
Missing a docstring and/or is this private (not public) for some reason?

Reply:
No docstring because it isn't yet documented on our site. Ticket is in our backlog to add. I added it here though.